ocpp-v1_6/src/main/java/eu/chargetime/ocpp/WebSocketTransmitter.java

```diff
@@ -93,6 +96,18 @@ public void onError(Exception ex)
             	}
             }
         };
+
+        if(WSS_SCHEME.equals(resource.getScheme())) {
+            try {
+                client.setSocket(wssSocketBuilder
```

Comment:
I haven't tested this theory, but this could end in a null pointer exception.
If the url starts with "wss:" and I didn't call enableWSS(...) before I called connect(...)
Then the wssSocketBuilder is null --> Null pointer exception.
Instantiating the wssSocketBuilder in the constructor would work around this, but I can't tell what would happen if I tried to connect without an sslContext.

Reply:
I've added verification to SSL... builders to fail fast if critical parameters we know at the moment of server/client creation are missing. Also created separate constructor for WSS scheme vs old constructor to be used for WS. I suggest a WS-initialized client must not (due to security reasons) try to communicate via WSS scheme and it probably won't be able to establish a connection in practice if a server is configured properly. So I've added explicit IllegalStateException in case we try to connect WS-initialized client to WSS endpoint.